### PR TITLE
Add version command and global version flag

### PR DESCRIPTION
**Pull request by jfmyers9**
This change adds a `bpm version` command and a global `--version` flag
that can be used to display the version of bpm. The `src/version` file
controls the version displayed by this command when created deployed in
a BOSH release. The final release build will bump the value of this file
to the version that is being created, create a final release, and then
add a `+dev` to the currently released version.

[finishes #155400037]

### CODE REVIEW

File: src/bpm/integration/version_test.go [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+			session, err := gexec.Start(command, GinkgoWriter, GinkgoWriter)
+			Expect(err).ShouldNot(HaveOccurred())
+			Eventually(session).Should(gexec.Exit(0))
+			Expect(session.Out).To(gbytes.Say("[DEV BUILD]"))
+		})
+
+		Context("when it is compile with a version", func() {
+			BeforeEach(func() {
+				var err error
+				bpmPath, err = gexec.Build("bpm/cmd/bpm", "-ldflags", "-X bpm/commands.Version=1.2.3")
```
Comment (xoebus) [inline src/bpm/integration/version_test.go]: I like that this test will catch the failure if we change the name or package of the variable.

File: src/bpm/commands/version.go [hunk 2]
```diff
@@ -0,0 +1,48 @@
+// Copyright (C) 2017-Present CloudFoundry.org Foundation, Inc. All rights reserved.
```
Comment (xoebus) [inline src/bpm/commands/version.go]: Nit: 2018 etc.

**Comment by aramprice** [comment]
would be nice to fix shell-check errors for `ci/scripts/bump-local-version`, eventually. looks good otherwise.

File: src/bpm/commands/version.go (hunk 2 above)
Comment (jfmyers9) [reply src/bpm/commands/version.go]: Are we supposed to update this everything we touch a file?

**Comment by jfmyers9** [comment]
Fixed shellcheck complaints in the `bump-local-version` script.

**Comment by xoebus** [comment]
Can we get shellcheck running in CI?

File: src/bpm/commands/version.go (hunk 2 above)
Comment (xoebus) [reply src/bpm/commands/version.go]: No, just new files.

**Pull request merged**